cpu_driver/src/api/cpu_driver.c

```diff
@@ -0,0 +1,144 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <api/cpu_driver.h>
+#include <errno.h>
+#include <string.h>
+
+/** Inlude the implementated functions */
+#include <api/driver/i_cpu_driver.h>
+
+
+/**
+ * Initialize the papiGPU
+ */
+int papiGPU_initialize(gpu_portname         portname[],
+                       enum papiGPU_states *state)
+{
+  int status = 0;
+
+  // Check for valid arguments
+  if (NULL == state){
+    #ifdef DEBUGLOG
+    printf ("\x1B[31m" "ERROR: " "\x1B[0m" "The state pointer is null. " \
+            "Error code: %d\n", EINVAL);
+    #endif
+    return EINVAL;
+  }
+
+  *state = GPU_ERROR;
+
+  status = strcmp(portname, "");
+  if (!status){
+    #ifdef DEBUGLOG
+    printf ("\x1B[31m" "ERROR: " "\x1B[0m" "The GPU portname is empty. " \
+            "Error code: %d\n", EINVAL);
+    #endif
+    *state = GPU_ERROR;
+    return EINVAL;
+  }
+
+  status = i_papiGPU_initialize(portname, state);
+
+  if (!status){
+    #ifdef DEBUGLOG
+    printf ("\x1B[32m" "INFO: " "\x1B[0m" "The GPU was initialized.\n");
+    #endif
+    *state = GPU_INITIALIZED;
+  }
+
+  return status;
+}
+
+
+/**
+ * Create the camera
+ */
+int papiGPU_create_camera(struct papiGPU_vertex  cam_vertex,
+                          gpu_focal_point        fp_distance,
+                          enum papiGPU_states   *state)
+{
+  int status = 0;
+
+  // Check for valid arguments
+  if (NULL == state){
+    #ifdef DEBUGLOG
+    printf ("\x1B[31m" "ERROR: " "\x1B[0m" "The state pointer is null. " \
```

Comment:
May want to replace printf with a function that prints into stderr. Aslo may want to have features allowing logging errors into a file